### DISCUSSION TITLE
Restrict width of notes to table width in LaTeX output

### DISCUSSION THREAD

**Discussion by turbanisch** [rstudio/gt#860] [Ideas]
Dear community - I'm relatively new to `gt` but already in love with its design philosophy. I understand that html output is the focus of this package yet some functionality to create LaTeX output is provided. 

It seems that the current implementation simply creates a minipage environment of line width for table notes. In many academic publications, however, table notes extend no further than the width of the table itself. Should this option be added (or even set as the default)? Although not perfect by themselves, the LaTeX packages `threeparttable` and `threeparttablex` could be used for this.

Below I include a sample R Markdown document illustrating my case:
````
---
title: "GT LaTeX"
output: 
  pdf_document:
    keep_tex: true
editor_options: 
  chunk_output_type: console
---

```{r, include=FALSE}
library(tidyverse)
library(gt)
```

```{r}
df <- mtcars %>% head(5) %>% select(1:5)

df %>% 
  gt() %>% 
  tab_source_note(md("*Note:* This is a very long note that needs to be wrapped. This is a very long note that needs to be wrapped. This is a very long note that needs to be wrapped.")) %>% 
  tab_source_note("Here is another one.") %>% 
  tab_footnote(
    footnote = md("medium-sized **cylinders**"),
    locations = cells_body(columns = cyl, rows = (cyl == 6))
  )
```
````

yielding 

<img width="655" alt="grafik" src="https://user-images.githubusercontent.com/51969412/147609539-9b994fa8-bc32-42d8-bcfc-f506807bfc3b.png">

And the version using `threeparttablex`:
<img width="220" alt="grafik" src="https://user-images.githubusercontent.com/51969412/147609828-e90e45e7-bcf7-4aa1-89be-fc0bf7552659.png">

**Comment by turbanisch** [comment]
Another benefit of using `threeparttable` would be that it has a special command (`\tnote`) for creating superscript references to footnotes in the table body that won't ruin the alignment of the actual cell contents (unlike in the example above).

**Comment by rich-iannone** [comment]
I agree that the minipage usage for the footer is not ideal and this is one of those cases where the idea breaks down. I am pretty impressed with `threeparttable` and I do see the benefits of moving towards that. My remaining concerns with that are the continuing maintenance of that package and having that as a default package in tinytex (but I guess it just gets downloaded anyway?).

**Comment by rich-iannone** [comment]
This seems like a pretty reasonable feature request, so I'll move this discussion into an issue.

**Comment by turbanisch** [comment]
Thanks for turning this into an issue! I implemented the `threeparttable` environment in a fork of this package a while ago and it works as expected. Compared to the current version, however, we would lose the ability to break tables across multiple pages. This feature is currently provided by the `longtable` environment whose syntax differs considerably from that of `threeparttable`. 

I went down the rabbit hole of LaTeX table environments a bit and it seems to be a recurring issue that certain environments implement a certain feature (such as table footnotes) quite well, but at the cost of being incompatible with others. 

The only package that unites *all* of these features in a common syntax and is flexible enough to be a proper match of `GT`'s functionality is the package [`tabularray`](https://mirror.physik.tu-berlin.de/pub/CTAN/macros/latex/contrib/tabularray/tabularray.pdf), in my opinion. It is still under active development though, so that it might not be the prefered choice for all users. I'll try to implement it for personal use nonetheless because I am convinced that it already surpasses all other table packages in LaTeX.
